src/cli/cli_coap.cpp

```diff
@@ -37,6 +37,7 @@
 
 #include <openthread/random_noncrypto.h>
 
+#include <cstddef>
```

Comment:
Do we need to include this? Can we remove this?

Reply:
It was only needed because of compilation failures with macOS target on _size_t_ types I used in `static_cast<size_t>()` type casts . But this could be removed since I considered your next suggestion a better solution so all these type casts will disappear therefore this include is not needed anymore.